addons/hr_holidays/tests/test_leave_requests.py

```diff
@@ -1400,12 +1400,17 @@ def test_public_holiday_in_the_middle_of_flexible_request(self):
             'date_to': datetime(2022, 3, 11, 23, 59, 59),
             'calendar_id': calendar.id,
         })
+        # Create a public holiday with no calendar
+        self.env['resource.calendar.leaves'].create({
+            'date_from': datetime(2022, 3, 12),
+            'date_to': datetime(2022, 3, 12, 23, 59, 59),
+        })
 
         leave = self.env['hr.leave'].with_user(self.user_employee_id).create({
             'name': 'Holiday Request',
             'employee_id': self.employee_emp.id,
             'holiday_status_id': self.holidays_type_1.id,
             'request_date_from': date(2022, 3, 10),
-            'request_date_to': date(2022, 3, 12),
+            'request_date_to': date(2022, 3, 13),
         })
         self.assertEqual(leave.number_of_days, 2)
```

Comment:
Maybe try to create 2 leaves instead of just one, to ensure that without the presence of a public holiday, the leave has a correct duration as well. You could also create 2 public holidays with one being for a different company than the employee is in